blaze/compute/sql.py

```diff
@@ -487,18 +495,27 @@ def compute_up(t, s, **kwargs):
     elif isinstance(t.apply, Summary):
         reduction = compute(t.apply, {t._child: s}, post_compute=False)
 
-    # d = dict((c.name, c) for c in inner_columns(s))
-    # grouper = [d[col] for col in t.grouper.fields]
     grouper = [lower_column(s.c.get(col)) for col in t.grouper.fields]
 
-    s2 = reduction.group_by(*grouper)
-
-    for g in grouper:
-        s2.append_column(g)
-
-    cols = list(s2.inner_columns)
-    cols = cols[-len(grouper):] + cols[:-len(grouper)]
-    return s2.with_only_columns(cols)
+    grouper_columns = pipe(grouper, map(get_inner_columns), concat)
+    reduction_columns = pipe(reduction.inner_columns, map(get_inner_columns),
+                             concat)
+    columns = list(unique(chain(grouper_columns, reduction_columns)))
+    if (not isinstance(s, sa.sql.selectable.Alias) or
+            (hasattr(s, 'froms') and isinstance(s.froms[0],
+                                                sa.sql.selectable.Join))):
+        from_obj = s.froms[0]
+    else:
+        from_obj = None
+    return sa.select(columns=columns,
+                     whereclause=getattr(s, 'element', s)._whereclause,
+                     from_obj=from_obj,
+                     distinct=getattr(s, 'element', s)._distinct,
+                     group_by=grouper,
+                     having=None,  # TODO: we don't have an expression for this
```

Comment:
names and account totals for people who have more than $1000 in their accounts

```sql
select name, sum(amount) as total from accounts group by name having sum(amount) > 1000
```

blaze could do pretty much the same thing syntactically:

```python
by(t.name, total=t.amount.sum()).having(t.amount.sum() > 1000)
```